api/envoy/config/core/v3/config_source.proto

```diff
@@ -149,10 +150,28 @@ message RateLimitSettings {
 // <arch_overview_service_discovery>` etc. may either be sourced from the
 // filesystem or from an xDS API source. Filesystem configs are watched with
 // inotify for updates.
-// [#next-free-field: 8]
+// [#next-free-field: 9]
 message ConfigSource {
   option (udpa.annotations.versioning).previous_message_type = "envoy.api.v2.core.ConfigSource";
 
+  // This is the same as TypedExtensionConfig. However, extensions.proto cannot be
+  // imported as it defines a message (ExtensionConfigSource) that depends on ConfigSource.
+  // [#next-major-version: refactor the file envoy/config/core/v3/extension.proto
```

Comment:
@adisuissa you are free to move protos between files in the same directory/package. This does not change any generated code or wire level artifacts. It might break folks code-wise during import, but not beyond include paths, and we don't have a strong version policy on that. I think it would be the lesser evil in this case.

Reply:
I was under the impression that it changes the Jave file name (`option java_outer_classname = "ConfigSourceProto";` will not hold now), but if that's not important, I agree with you that splitting is the right thing to do.

Reply:
You're probably right, nonetheless we don't explicitly rule this out in https://github.com/envoyproxy/envoy/blob/main/api/API_VERSIONING.md and have historically done this without any major complaints. If you want to lock this down, as in principle it is inconsistent with the "don't break consuming code" notion, that's fine, but then update style guide. I'm very much on the fence tilting towards not doing that, since there is no problem flagged to fix.